### ISSUE TITLE
[REVIEW]: SIHR: a MATLAB/GNU Octave toolbox for single image highlight removal

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1822]
**Submitting author:** @vitorsr (<a href="http://orcid.org/0000-0002-7583-5577">Vítor Saraiva Ramos</a>)
**Repository:** <a href="https://github.com/vitorsr/SIHR" target ="_blank">https://github.com/vitorsr/SIHR</a>
**Version:** v0.1
**Editor:** @Kevin-Mattheus-Moerman
**Reviewer:** @thomakra, @Atif-Anwer
**Archive:** <a href="https://doi.org/10.5281/zenodo.3367760" target="_blank">10.5281/zenodo.3367760</a>

## Status

[![status](https://joss.theoj.org/papers/179c16c6e52e7d594ebc99fd6a1103d7/status.svg)](https://joss.theoj.org/papers/179c16c6e52e7d594ebc99fd6a1103d7)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/179c16c6e52e7d594ebc99fd6a1103d7"><img src="https://joss.theoj.org/papers/179c16c6e52e7d594ebc99fd6a1103d7/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/179c16c6e52e7d594ebc99fd6a1103d7/status.svg)](https://joss.theoj.org/papers/179c16c6e52e7d594ebc99fd6a1103d7)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/vitorsr/SIHR" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@thomakra & @Atif-Anwer, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @Kevin-Mattheus-Moerman know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/vitorsr/SIHR">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@vitorsr) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/vitorsr/SIHR">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@vitorsr) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @thomakra, @Atif-Anwer it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01822/joss.01822/10.21105.joss.01822.pdf)

**Comment by Kevin-Mattheus-Moerman** [comment]
@vitorsr this is where the review takes place. 

@thomakra, @Atif-Anwer, thanks for helping to review this work. The checkboxes at the top guide you through the review process. You can make comments here and/or create dedicated issues on the software's repo and link to them here. 

Please let me know if you have any questions.

**Comment by thomakra** [comment]
@Kevin-Mattheus-Moerman I've gone though the checklist. Most things seem to be in order. I've added a few issues in the repo with small suggestions for improving the documentation.
One thing I noted was that the Zenodo DOI in the paper seems to link to a removed entry and is not the same as the one linked to in the README.

I can't find any automated tests, but number of methods are small enough that I could manually test them out on the linked test images. Most seem to work fine, but I've added an issue with an Octave incompatibility, which the author is looking into.

Since this is my first review, please give me a hint if there's anything further you require from my side.

**Comment by vitorsr** [comment]
Hi, first of all, thank you @thomakra and @Atif-Anwer for taking the time to review this submission, and @Kevin-Mattheus-Moerman for editing.

I have acted according to the feedback given by @thomakra - the [latest commit](https://github.com/vitorsr/SIHR/commit/397534a3c1c866a09dc9eba66424a4eb0aa217f4) incorporates the suggestions given and closes the corresponding issues.

Please, feel free to re-open them if any change did not apply, or create a new one for more suggestions.

**Comment by vitorsr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01822/joss.01822/10.21105.joss.01822.pdf)

**Comment by Atif-Anwer** [comment]
@Kevin-Mattheus-Moerman Apologies for a slightly delayed review. I've reviewed the repository, the paper and gone though the checklist. Overall the paper and repository are well maintained and well written. I've added an [issue ](https://github.com/vitorsr/SIHR/issues/11#issue-513811018) in the repo with minor suggestions and QoL changes.

I successfully reproduced the claimed results in Matlab (2018b) but do not have access to or experience in Octave. However since that is already covered by some fine work by the other reviewer, it is more than satisfactory. 
This is my first review for JOSS, so If the review is lacking in some places then do let me know. 
Thanks and regards!

**Comment by Kevin-Mattheus-Moerman** [comment]
@vitorsr can you comment on/deal with @Atif-Anwer 's feedback/issues?

**Comment by Kevin-Mattheus-Moerman** [comment]
@thomakra can you review the updates implemented by @vitorsr ?

**Comment by vitorsr** [comment]
> @vitorsr can you comment on/deal with @Atif-Anwer 's feedback/issues?

@Kevin-Mattheus-Moerman, I have noted @Atif-Anwer's concerns and am currently working on pushing changes that incorporate his feedback to the repository. Whenever I do so, I will ping you and the reviewers in this thread.

In the meantime, if @thomakra has additional comments on the updates, I'll be happy to incorporate them as well.

**Comment by Kevin-Mattheus-Moerman** [comment]
@vitorsr any updates on these developments?

**Comment by thomakra** [comment]
Sorry for not following up.
I've been moving house and I'm a little behind on a lot of things.
I'll have another look tomorrow.

-Thomas-

**Comment by vitorsr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01822/joss.01822/10.21105.joss.01822.pdf)

**Comment by vitorsr** [comment]
> @vitorsr any updates on these developments?

Yes! I have pushed changes that implemented all feedback given by @Atif-Anwer. Since it also had suggestions regarding the paper, I have re-generated a new proof.

**Comment by Atif-Anwer** [comment]
I've gone through the updated manuscript as well as the changes committed in the repo. All the suggestions I mentioned via issue [#11](https://github.com/vitorsr/SIHR/issues/11) have been addressed to. I've also completed the review checklist above since all the issues have been verified at my end. Is there any other thing required regarding the review process from my side now? @Kevin-Mattheus-Moerman

**Comment by Kevin-Mattheus-Moerman** [comment]
@Atif-Anwer no all set from your end. Thanks a lot for your help! :tada:

**Comment by Kevin-Mattheus-Moerman** [comment]
@thomakra thanks also for your review efforts. You have a couple of boxes unchecked. Have you clarified to @vitorsr what the main issues are? Thanks.

**Comment by thomakra** [comment]
I've had a look at the latest changes and they seem to address the provided suggestions.
I was a bit unsure about what to answer under *State of the field* as I don't think the author describes how this software compares to other commonly-used packages although the paper references relevant literature on the subject, hence I have left this unchecked.

Other than that I have no further suggestions.

**Comment by vitorsr** [comment]
> @thomakra thanks also for your review efforts. You have a couple of boxes unchecked. Have you clarified to @vitorsr what the main issues are? Thanks.

@Kevin-Mattheus-Moerman I'd like to note that I have pushed some [changes](https://github.com/vitorsr/SIHR/commit/397534a3c1c866a09dc9eba66424a4eb0aa217f4) aimed at addressing the concerns in the issues raised by @thomakra.

> I was a bit unsure about what to answer under State of the field as I don't think the author describes how this software compares to other commonly-used packages although the paper references relevant literature on the subject, hence I have left this unchecked.

@thomakra I believe I address this in the [README.md](https://github.com/vitorsr/SIHR#raison-d%C3%AAtre) through the raison d'être (French equivalent to “statement of purpose”) subsection by linking to the source code of some of the implemented methods which were ported to MATLAB/Octave, and in the summary paper by noting that this problem has scarce software availability.

As in, there are no software packages yet on the problem except for some source code provided by some authors to reproduce their results.

**Comment by thomakra** [comment]
I apologize, @vitorsr. I agree, this is well addressed in the readme file. I have checked the final box.

**Comment by vitorsr** [comment]
Hi @Kevin-Mattheus-Moerman, is there anything else due from my end?

**Comment by Kevin-Mattheus-Moerman** [comment]
@vitorsr no. I think this looks good.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01822/joss.01822/10.21105.joss.01822.pdf)

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1002/col.5080100409 is OK
- 10.1007/978-0-387-31439-6_538 is OK
- 10.1109/icip.2017.8297078 is OK
- 10.1111/j.1467-8659.2011.01971.x is OK
- 10.1007/978-3-319-56010-6_17 is OK
- 10.5281/zenodo.3367760 is OK
- 10.1109/tpami.2005.36 is OK
- 10.1109/icip.2006.312650 is OK
- 10.1016/j.patcog.2008.01.026 is OK
- 10.1364/ao.48.002711 is OK
- 10.1007/978-3-642-15561-1_7 is OK
- 10.1364/ao.52.004483 is OK
- 10.1016/j.cviu.2015.09.001 is OK
- 10.3169/mta.7.92 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@vitorsr 
- [ ] Your paper is about to be processed for acceptance. Can you double check the paper yourself too? 
- [ ] Can you check and extend you affiliation to include the city and country? 
- [ ] Can you rephrase the following `...starting from the most performing ones...`? Perhaps use something like this instead (feel free to disagree and propose an alternative form): `...starting with the most computationally efficient ones...`
- [ ] Change `As of time of writing, the following methods are available.` to `At the time of writing the methods listed in Table 1 are available.` 
- [ ] The final paper will contain a link to the latest archived version. Therefore please remove this sentence: `The source code for SIHR is being archived by Zenodo since its pre-release version (Ramos, 2019).`(you can also clean up the .bib file when this reference is removed).

**Comment by vitorsr** [comment]
@Kevin-Mattheus-Moerman, thanks for the additions. I have incorporated all in the latest commit. Also,  I have double-checked the paper, and it is sound on my end.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01822/joss.01822/10.21105.joss.01822.pdf)

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept

**Comment by whedon** [comment]
No archive DOI set. Exiting...

**Comment by Kevin-Mattheus-Moerman** [comment]
@vitorsr at this point can you please archive the software on Zenodo and report back with the DOI of the archived version here? 
Please make sure the meta data for the archived version (i.e. the authors and title) match that of the JOSS paper.

**Comment by vitorsr** [comment]
@Kevin-Mattheus-Moerman the version v0.1 is still accurate. The DOI of the archived version is [10.5281/zenodo.3367760](https://doi.org/10.5281/zenodo.3367760) (all versions, resolves to the latest archive.)

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon set 10.5281/zenodo.3367760 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3367760" target="_blank">10.5281/zenodo.3367760</a> is the archive.

**Comment by Kevin-Mattheus-Moerman** [comment]
@vitorsr please amend the archive title to match the title of your paper

**Comment by vitorsr** [comment]
Fixed. Sorry for the oversight.

**Comment by Kevin-Mattheus-Moerman** [comment]
@openjournals/joss-eics this submission is ready for acceptance

**Comment by kyleniemeyer** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01822/joss.01822/10.21105.joss.01822.pdf)

**Comment by kyleniemeyer** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1002/col.5080100409 is OK
- 10.1007/978-0-387-31439-6_538 is OK
- 10.1109/icip.2017.8297078 is OK
- 10.1111/j.1467-8659.2011.01971.x is OK
- 10.1007/978-3-319-56010-6_17 is OK
- 10.1109/tpami.2005.36 is OK
- 10.1109/icip.2006.312650 is OK
- 10.1016/j.patcog.2008.01.026 is OK
- 10.1364/ao.48.002711 is OK
- 10.1007/978-3-642-15561-1_7 is OK
- 10.1364/ao.52.004483 is OK
- 10.1016/j.cviu.2015.09.001 is OK
- 10.3169/mta.7.92 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by kyleniemeyer** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1002/col.5080100409 is OK
- 10.1007/978-0-387-31439-6_538 is OK
- 10.1109/icip.2017.8297078 is OK
- 10.1111/j.1467-8659.2011.01971.x is OK
- 10.1007/978-3-319-56010-6_17 is OK
- 10.1109/tpami.2005.36 is OK
- 10.1109/icip.2006.312650 is OK
- 10.1016/j.patcog.2008.01.026 is OK
- 10.1364/ao.48.002711 is OK
- 10.1007/978-3-642-15561-1_7 is OK
- 10.1364/ao.52.004483 is OK
- 10.1016/j.cviu.2015.09.001 is OK
- 10.3169/mta.7.92 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1219

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1219, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kyleniemeyer** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1217134548724658177) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1220
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01822](https://doi.org/10.21105/joss.01822)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by kyleniemeyer** [comment]
Congrats @vitorsr on your article's publication in JOSS!

Many thanks to @thomakra and @Atif-Anwer for reviewing, and @Kevin-Mattheus-Moerman for editing, this submission.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01822/status.svg)](https://doi.org/10.21105/joss.01822)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01822">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01822/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01822/status.svg
   :target: https://doi.org/10.21105/joss.01822
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01822/status.svg)](https://doi.org/10.21105/joss.01822)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
